fuzzer/src/cairo_compiled_programs_fuzzer.rs

```diff
@@ -0,0 +1,541 @@
+
+    let program_content_array_sum = std::fs::read(&json_path_array_sum).unwrap();
+
+    // Run the program with default configurations
+    cairo_run::cairo_run(&program_content_array_sum, cairo_run_config, hint_executor)
+        .expect("failed to run program");
+
+    // Remove files to save memory
+    delete_files(&cairo_path_array_sum, &json_path_array_sum);
+}
+
+fn program_unsafe_keccak(
+    array: String,
+    cairo_run_config: &CairoRunConfig,
+    hint_executor: &mut BuiltinHintProcessor,
+) {
+    let populated_array = array
+        .chars()
+        .enumerate()
+        .map(|(index, num)| {
+            format!(
+                "assert data[{}] = {}; \n",
+                index,
+                num.to_string().repeat(array.len() - (index + 1))
+            )
+        })
+        .collect::<Vec<_>>()
+        .join("            ");
+
+    let file_content = format!(
+        "
+    %builtins output
+
+    from starkware.cairo.common.alloc import alloc
+    from starkware.cairo.common.serialize import serialize_word
+    from starkware.cairo.common.keccak import unsafe_keccak
+
+    func main{{output_ptr: felt*}}() {{
+        alloc_locals;
+
+        let (data: felt*) = alloc();
+
+        {}
+
+        let (low: felt, high: felt) = unsafe_keccak(data, {});
+
+        serialize_word(low);
+        serialize_word(high);
+
+        return ();
+    }}
+    ",
+        populated_array,
+        array.len()
+    );
+
+    // Create programs names and program
+    let cairo_path_unsafe_keccak = format!(
+        "cairo_programs/unsafe_keccak_{:?}.cairo",
+        FUZZ_ITERATION_COUNT
+    );
+    let json_path_unsafe_keccak = format!(
+        "cairo_programs/unsafe_keccak_{:?}.json",
+        FUZZ_ITERATION_COUNT
+    );
+    let _ = fs::write(&cairo_path_unsafe_keccak, file_content.as_bytes());
+
+    compile_program(&cairo_path_unsafe_keccak, &json_path_unsafe_keccak);
+
+    let program_content_unsafe_keccak = std::fs::read(&json_path_unsafe_keccak).unwrap();
+
+    // Run the program with default configurations
+    cairo_run::cairo_run(
+        &program_content_unsafe_keccak,
+        cairo_run_config,
+        hint_executor,
+    )
+    .expect("failed to run program");
+
+    // Remove files to save memory
+    delete_files(&cairo_path_unsafe_keccak, &json_path_unsafe_keccak);
+}
+
+fn program_bitwise(
+    num1: u8,
+    num2: u8,
+    cairo_run_config: &CairoRunConfig,
+    hint_executor: &mut BuiltinHintProcessor,
+) {
+    let and = num1 & num2;
+    let xor = num1 ^ num2;
+    let or = num1 | num2;
+
+    let file_content = format!("
+    %builtins bitwise
+    from starkware.cairo.common.bitwise import bitwise_and, bitwise_xor, bitwise_or, bitwise_operations
+    from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+
+    func main{{bitwise_ptr: BitwiseBuiltin*}}() {{
+        let (and_a) = bitwise_and({}, {});  
+        assert and_a = {}; 
+        let (xor_a) = bitwise_xor(12, 10);
```

Comment:
You can embed the variable name inside the format string:

```suggestion
        let (and_a) = bitwise_and({num1}, {num2});
        assert and_a = {and};
        let (xor_a) = bitwise_xor({num1}, {num2});
```

Also, I think the arguments to `bitwise_xor` and `bitwise_or` should be `{num1}, {num2}` instead of `12, 10`